### PR TITLE
Update Android SDK target to 14

### PR DESCRIPTION
**Pull request by wec43**
Close https://github.com/tutao/private-issues/issues/207

### CODE REVIEW

File: app-android/app/build.gradle [hunk 1]
```diff
@@ -155,45 +154,44 @@ dependencies {
...
 		implementation fileTree(include: ['*.aar'], dir: '../libs')
 	} else {
 		implementation "net.zetetic:android-database-sqlcipher:4.5.0"
 	}
-	implementation "androidx.sqlite:sqlite:2.0.1"
+	implementation 'androidx.sqlite:sqlite-ktx:2.4.0'
 
-	implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.4.1'
+	implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.8.3'
 	implementation "androidx.lifecycle:lifecycle-livedata-ktx:$lifecycle_version"
```
Comment (charlag) [inline app-android/app/build.gradle:170]: we can throw away livedata! maybe!!

File: app-android/app/build.gradle [hunk 2]
```diff
@@ -132,18 +131,18 @@ tasks.withType(Test).configureEach {
...
-	def activity_version = "1.4.0"
-    def coroutines_version = "1.8.0"
+	def room_version = '2.6.1'
+	def lifecycle_version = '2.8.3'
+	def activity_version = '1.9.0'
+    def coroutines_version = '1.8.1'
 
 	implementation "de.tutao:tutasdk"
 
 	// Important: cannot be updated without additional measures as Android 6 and 7 do not have Java 9
```
[the remark this answers is not in the document]
Comment (wec43) [reply app-android/app/build.gradle]: Removed.

File: app-android/app/build.gradle [hunk 3]
```diff
@@ -132,18 +131,18 @@ tasks.withType(Test).configureEach {
 
 
 dependencies {
-	def room_version = "2.4.2"
-	def lifecycle_version = "2.4.1"
-	def activity_version = "1.4.0"
-    def coroutines_version = "1.8.0"
+	def room_version = '2.6.1'
```
[the remark this answers is not in the document]
Comment (wec43) [reply app-android/app/build.gradle:135]: I added an option to let KSP generate Kotlin in gradle.

File: app-android/app/build.gradle (hunk 1 above)
Comment (wec43) [reply app-android/app/build.gradle]: Yes it is removed.

**Review by charlag — APPROVED** [review]
Please prefix both commits with `[android]`

**Pull request merged**